processing/src/main/java/io/druid/query/Druids.java

```diff
@@ -949,6 +953,7 @@ public SegmentMetadataQueryBuilder copy(SegmentMetadataQueryBuilder builder)
           .dataSource(builder.dataSource)
           .intervals(builder.querySegmentSpec)
           .toInclude(toInclude)
+          .analysisTypes(analysisTypes.toArray(new SegmentMetadataQuery.AnalysisType[analysisTypes.size()]))
```

Comment:
is it required to set analysisTypes before calling this ? If not this will throw an NPE.